### ISSUE TITLE
ScriptHookVDotNet causes the game to run on background after closing

### ISSUE THREAD

**Issue by seaque** [crosire/scripthookvdotnet#1063]
**Version**
SHVDN v3.4.0

**Description**
After closing the game gta5.exe still runs on background, using roughly 2 GB of memory in a system with 16 GB RAM. Deleting SHVDN dlls from the game directory solves it, so the problem should be about SHVDN.

**Platform**
Windows 10 21H2

**Comment by WithLithum** [comment]
Experienced this problem, and have to manually shut Rockstar Games Launcher down every time after closing the game, except for when RPH was hooked.

**Comment by gameplayoffert** [comment]
I'm on windows 8.1 and it happen for me too and it is very very very annoying.

**Comment by kagikn** [comment]
Actually, I have this problem when SHVDN is loaded, too. No idea how to solve it yet, though.

**Comment by seaque** [comment]
It seems that it appears starting from SHVDN 3.0. I've tried multiple versions (3.4, 3.1, 3.0) without any mods and only with 2.10.10 i didn't have a problem and the game closed completely.

**Comment by kagikn** [comment]
As I casually tested, I think the bug lies in `DllMain.cop. When I change the code to prevent the asi from even loading `SHVDN.ScriptDomain`, the process can exit properly.

**Comment by kagikn** [comment]
Man, not good. Looks like the problem causes from this line, but I have no idea how to fix it.
https://github.com/crosire/scripthookvdotnet/blob/ccdf0c1b13688b52c36cf3322da1d15eaea7bc4c/source/core/ScriptDomain.cs#L163

**Comment by Dave247** [comment]
This has also long been an issue even without any mods, apparently due to a bug with you have the Logitech G Hub program installed/running, so make sure it's not that interfering.
https://www.reddit.com/r/GrandTheftAutoV_PC/comments/mgj7fx/gta_wont_shut_down_properly_stays_running_on/

**Comment by cpast** [comment]
> Man, not good. Looks like the problem causes since this statement executes, but I have no idea how to fix it.
> 
> https://github.com/crosire/scripthookvdotnet/blob/ccdf0c1b13688b52c36cf3322da1d15eaea7bc4c/source/core/ScriptDomain.cs#L163

Dumb thought: try hooking something called on game exit, and tearing down all app domains when that is called?

**Comment by kagikn** [comment]
Closing since I'm sure #1167 solved this one.

**Comment by nomakewan** [comment]
Just as a note, I actually did have GTA5.exe hang in the background after quitting the game and need to be terminated manually with SHVDN e0d6d35 running. So while it has only happened once so far, that seems to suggest that it did not in fact resolve the issue.

**Comment by Sardelka9515** [comment]
It never happened to me, does it present without rockstar launcher?

**Comment by nomakewan** [comment]
I'm on Steam and launch from Steam. I generally notice because I'll quit the game and go to make a report about things on Discord, but I'll notice that the Discord in-game notification saying I'm playing Grand Theft Auto V will still be there. Then sure enough, GTA5.exe is just chilling in the process list eating up several gigs of RAM. Kill that process and that's that. It's done that since as far back as I can remember (I started messing with GTAV in February 2022).

I tested this latest build specifically to make sure the fiber/threading changes didn't do anything funky with our mod and was indeed hoping that the changes would resolve the hang-in-background bug. But alas, didn't work out.

I will say that it happens very rarely, and I have not found a way to force it to happen. All I can say is that even with the latest SHVDN built from source, it is indeed possible to still happen.

**Comment by nomakewan** [comment]
Just an update that while again, this is exceedingly rare, I did manage to just have it happen again. I'm using bdbd841 and the one change I did make is I was testing out the Resource Adjuster mod, which apparently includes some stuff from FiveM to raise certain resource limits and make better use of VRAM.

After a few hours ingame I exited and noticed that my Discord notification was still there, and sure enough GTA5.exe was hanging with about 4.5 GB of RAM usage. Had to kill the process manually.

SHVDN or Resource Adjuster? Incompatibility between the current SHVDN and whatever patches Resource Adjuster is using? No clue. But figured I would mention it.
